includes/validation/class-amp-validation-manager.php

```diff
@@ -2135,6 +2124,45 @@ public static function validate_url( $url ) {
 		);
 	}
 
+	/**
+	 * Validate URL and store result.
+	 *
+	 * @param string      $url  URL to validate.
+	 * @param int|WP_Post $post The amp_validated_url post to update. Optional. If empty, then post is looked up by URL.
+	 * @return WP_Error|array {
+	 *     Error on failure, or array on success.
+	 *
+	 *     @type int    $post_id          ID for the amp_validated_url post.
+	 *     @type array  $results          Validation results, where each nested array contains an error key and sanitized key.
+	 *     @type string $url              Final URL that was checked or redirected to.
+	 *     @type array  $queried_object   Queried object, including keys for 'type' and 'id'.
+	 *     @type array  $stylesheets      Stylesheet data.
+	 *     @type string $php_fatal_error  PHP fatal error which occurred during validation.
+	 * }
+	 */
+	public static function validate_url_and_store( $url, $post = null ) {
+		$validity = self::validate_url( $url );
+		if ( $validity instanceof WP_Error ) {
+			return $validity;
+		}
+
+		$args = wp_array_slice_assoc( $validity, [ 'queried_object', 'stylesheets', 'php_fatal_error' ] );
+		if ( $post ) {
+			$args['invalid_url_post'] = $post;
+		}
+
+		$r = AMP_Validated_URL_Post_Type::store_validation_errors(
+			wp_list_pluck( $validity['results'], 'error' ),
+			$validity['url'],
+			$args
+		);
+		if ( $r instanceof WP_Error ) {
+			return $r;
+		}
+		$validity['post_id'] = $r;
```

Comment:
Abbreviations like this should be avoided.
```suggestion
		$result = AMP_Validated_URL_Post_Type::store_validation_errors(
			wp_list_pluck( $validity['results'], 'error' ),
			$validity['url'],
			$args
		);
		if ( $result instanceof WP_Error ) {
			return $result;
		}
		$validity['post_id'] = $result;
```